aptos-move/framework/aptos-stdlib/sources/cryptography/ristretto255_bulletproofs.move

```diff
@@ -114,12 +178,63 @@ module aptos_std::ristretto255_bulletproofs {
         )
     }
 
+    #[test_only]
+    /// Computes a range proof for a batch of values `vals`, each committed with the corresponding randomness in `rs`,
+    /// under the default Bulletproofs commitment key; see `pedersen::new_commitment_for_bulletproof`.
+    /// Returns a tuple containing the batch range proof and a vector of said commitments.
+    /// Only works for `num_bits` in `{8, 16, 32, 64}` and batch sizes (length of `vals` and `rs`) in `{1, 2, 4, 8, 16}`.
+    public fun prove_batch_range_pedersen(
+        vals: &vector<Scalar>, rs: &vector<Scalar>,
+        num_bits: u64, dst: vector<u8>): (RangeProof, vector<pedersen::Commitment>)
+    {
+        let vals = std::vector::map_ref(vals, |val| scalar_to_bytes(val));
+        let rs = std::vector::map_ref(rs, |r| scalar_to_bytes(r));
+
+        let (bytes, compressed_comms) = prove_batch_range_internal(
+            vals, rs,
+            num_bits, dst,
+            &ristretto255::basepoint(), &ristretto255::hash_to_point_base()
+        );
+        let comms = std::vector::map(compressed_comms, |compressed_comm| {
+            let comm = pedersen::new_commitment_from_bytes(compressed_comm);
+            std::option::extract(&mut comm)
+        });
+
+        (
+            RangeProof { bytes },
+            comms
+        )
+    }
+
+    #[test_only]
+    /// Computes a range proof for a batch of values `vals`, each committed with the corresponding randomness in `rs`,
+    /// using custom base points `val_base` and `rand_base`. Each commitment is of the form `val * val_base + r * rand_base`.
+    /// Returns a tuple containing the batch range proof and a vector of said commitments.
+    /// Only works for `num_bits` in `{8, 16, 32, 64}` and batch sizes (length of `vals` and `rs`) in `{1, 2, 4, 8, 16}`.
+    public fun prove_batch_range(
+        vals: &vector<Scalar>, rs: &vector<Scalar>,
+        val_base: &RistrettoPoint, rand_base: &RistrettoPoint,
+        num_bits: u64, dst: vector<u8>): (RangeProof, vector<pedersen::Commitment>) 
+    {
+        let vals = std::vector::map_ref(vals, |val| scalar_to_bytes(val));
+        let rs = std::vector::map_ref(rs, |r| scalar_to_bytes(r));
+
+        let (bytes, compressed_comms) = prove_batch_range_internal(vals, rs, num_bits, dst, val_base, rand_base);
+        let comms = std::vector::map(compressed_comms, |compressed_comm| {
+            let comm = pedersen::new_commitment_from_bytes(compressed_comm);
+            std::option::extract(&mut comm)
+        });
+
+        (
+            RangeProof { bytes },
+            comms
+        )
+    }
+
     //
     // Native functions
     //
-
-    /// Aborts with `error::invalid_argument(E_DESERIALIZE_RANGE_PROOF)` if `proof` is not a valid serialization of a
-    /// range proof.
+    
```

Comment:
I don't understand: why is this comment removed. The native will indeed abort with a deserialization error...

Reply:
Done